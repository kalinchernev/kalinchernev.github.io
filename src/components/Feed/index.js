// @flow strict
export { default } from './Feed';
