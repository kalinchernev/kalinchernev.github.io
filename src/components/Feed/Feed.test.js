// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Feed from './Feed';

describe('Feed', () => {
  const props = {
    edges: [
      {
        node: {
          fields: {
            slug: '/test_0',
            categorySlug: '/test_0',
            tagSlugs: ['/test-1', '/test-2'],
          },
          frontmatter: {
            date: '2016-09-01',
            description: 'test_0',
            category: 'test_0',
            tags: ['test-1', 'test-2'],
            title: 'test_0',
          },
          id: 'test-123',
          html: '<p>test</p>',
        },
      },
      {
        node: {
          fields: {
            slug: '/test_1',
            categorySlug: '/test_1',
            tagSlugs: ['/test-1', '/test-2'],
          },
          frontmatter: {
            date: '2016-09-01',
            description: 'test_1',
            category: 'test_1',
            tags: ['test-1', 'test-2'],
            title: 'test_1',
          },
          id: 'test-321',
          html: '<p>test</p>',
        },
      },
    ],
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Feed {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
