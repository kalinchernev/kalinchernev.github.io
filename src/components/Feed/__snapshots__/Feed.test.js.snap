// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders correctly 1`] = `
<div
  className="feed"
>
  <div
    className="feed__item"
  >
    <div
      className="feed__item-meta"
    >
      <time
        className="feed__item-meta-time"
        dateTime="September 1, 2016"
      >
        September 2016
      </time>
      <span
        className="feed__item-meta-divider"
      />
      <span
        className="feed__item-meta-category"
      >
        <a
          className="feed__item-meta-category-link"
          href="/test_0"
        >
          test_0
        </a>
      </span>
    </div>
    <h2
      className="feed__item-title"
    >
      <a
        className="feed__item-title-link"
        href="/test_0"
      >
        test_0
      </a>
    </h2>
    <p
      className="feed__item-description"
    >
      test_0
    </p>
    <a
      className="feed__item-readmore"
      href="/test_0"
    >
      Read
    </a>
  </div>
  <div
    className="feed__item"
  >
    <div
      className="feed__item-meta"
    >
      <time
        className="feed__item-meta-time"
        dateTime="September 1, 2016"
      >
        September 2016
      </time>
      <span
        className="feed__item-meta-divider"
      />
      <span
        className="feed__item-meta-category"
      >
        <a
          className="feed__item-meta-category-link"
          href="/test_1"
        >
          test_1
        </a>
      </span>
    </div>
    <h2
      className="feed__item-title"
    >
      <a
        className="feed__item-title-link"
        href="/test_1"
      >
        test_1
      </a>
    </h2>
    <p
      className="feed__item-description"
    >
      test_1
    </p>
    <a
      className="feed__item-readmore"
      href="/test_1"
    >
      Read
    </a>
  </div>
</div>
`;
